day_10/task_2/pipe_maze.cpp

```diff
@@ -270,3 +276,86 @@ void AttributedMaze::color_neighbor(const MazePoint& loop_point, const MazePoint
         }
     }
 }
+
+void AttributedMaze::print_states()
+{
+    for(const auto& row : maze_)
+    {
+        for(const auto& single_tile : row)
+        {
+            if(single_tile.state == State::Undefined)
+            {
+                std::cout << ".";
+            }
+            if(single_tile.state == State::LeftColor)
+            {
+                std::cout << "L";
+            }
+            if(single_tile.state == State::RightColor)
+            {
+                std::cout << "R";
+            }
+            if(single_tile.state == State::Loop)
+            {
+                std::cout << "X";
+            }
+
+        }
+        std::cout << std::endl;
+    }
+}
```

Comment:
[ optional ]

Deliver this function as separated commit, which is not neccesary part of the solution. So maybe DEBUG prefix could be fit. Of course such commit would need to include `#include <iostream` directive